### PR TITLE
Implement JSON Schema compatibility rule. Fixes #373

### LINKED ISSUE #373: Implement JSON Schema compatibility rule
See `https://github.com/confluentinc/schema-registry/pull/1289`

**Tasks**
- [ ] Analyze existing tools
- ...

### PR DESCRIPTION
**Pull request by jsenko**
This is an implementation of a JSON Schema compatibility checker.

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/rules/compatibility/jsonschema/JsonSchemaDiffLibrary.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * @author Jakub Senko <jsenko@redhat.com>
+ */
+public class JsonSchemaDiffLibrary {
+
+    public static final ObjectMapper MAPPER = new ObjectMapper();
+
+    static {
+        MAPPER.registerModule(new JsonOrgModule());
```
Comment (alesj) [inline app/src/main/java/io/apicurio/registry/rules/compatibility/jsonschema/JsonSchemaDiffLibrary.java:38]: Perhaps move MAPPER init also to this static block.

**Comment by alesj** [comment]
Year in copyright is 2019 ...

**Comment by alesj** [comment]
Looks good.
Never thought we would need so much code for a simple JSON stuff. :-)

**Comment by jsenko** [comment]
I've added comprehensive tests with fixes for discovered issues.
Created a separate issue to deal with the copyright year #426

File: app/src/main/java/io/apicurio/registry/rules/compatibility/jsonschema/diff/BooleanSchemaDiffVisitor.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    private final BooleanSchema original;
+
+    public BooleanSchemaDiffVisitor(DiffContext ctx, BooleanSchema original) {
+        this.ctx = ctx;
+        this.original = original;
+    }
+
+    @Override
+    public void visitBooleanSchema(BooleanSchemaWrapper schema) {
+        super.visitBooleanSchema(schema);
```
Comment (jsenko) [inline app/src/main/java/io/apicurio/registry/rules/compatibility/jsonschema/diff/BooleanSchemaDiffVisitor.java:39]: Note: Check may be missing here

File: pom.xml [hunk 3]
```diff
@@ -142,6 +142,8 @@
 
         <!-- JSON Schema Validator -->
         <medeia-validator-jackson.version>1.1.1</medeia-validator-jackson.version>
+        <org.everit.json.schema.version>1.12.1</org.everit.json.schema.version><!-- TODO unification -->
+		<jackson-datatype-json-org.version>2.10.3</jackson-datatype-json-org.version>
```
Comment (jsenko) [inline pom.xml:146]: Formatting?

File: pom.xml [hunk 4]
```diff
@@ -350,6 +352,16 @@
                 <artifactId>medeia-validator-jackson</artifactId>
                 <version>${medeia-validator-jackson.version}</version>
             </dependency>
+			<dependency>
```
Comment (jsenko) [inline pom.xml:355]: Formatting?

File: pom.xml (hunk 3 above)
Comment (EricWittmann) [reply pom.xml:146]: Yeah - it would be good for everyone to stop using tabs for indentation in pom.xml files.

**Pull request merged**